packages/cactus-plugin-keychain-aws-sm/src/test/typescript/integration/plugin-keychain-aws-sm.test.ts

```diff
@@ -0,0 +1,149 @@
+import test, { Test } from "tape-promise/tape";
+import { v4 as internalIpV4 } from "internal-ip";
+
+import {
+  Containers,
+  LocalStackContainer,
+  K_DEFAULT_LOCALSTACK_HTTP_PORT,
+} from "@hyperledger/cactus-test-tooling";
+
+import { v4 as uuidv4 } from "uuid";
+
+import { LogLevelDesc } from "@hyperledger/cactus-common";
+
+import {
+  IPluginKeychainAwsSmOptions,
+  PluginKeychainAwsSm,
+  AwsCredentialType,
+} from "../../../main/typescript/public-api";
+
+import fs from "fs";
+import path from "path";
+import os from "os";
+
+const logLevel: LogLevelDesc = "TRACE";
+
+test("get,set,has,delete alters state as expected", async (t: Test) => {
+  const localStackContainer = new LocalStackContainer({
+    logLevel: logLevel,
+  });
+  await localStackContainer.start();
+
+  const ci = await Containers.getById(localStackContainer.containerId);
+  const localstackIpAddr = await internalIpV4();
+  const hostPort = await Containers.getPublicPort(
+    K_DEFAULT_LOCALSTACK_HTTP_PORT,
+    ci,
+  );
+  const localstackHost = `http://${localstackIpAddr}:${hostPort}`;
+
+  test.onFinish(async () => {
+    await localStackContainer.stop();
+    await localStackContainer.destroy();
+  });
+
+  // Using awsCredentialType: AwsCredentialType.FromAwsCredentialFile
+  {
+    // Create aws credential file in a local directory
+    let tmpDirPath = "tmpDirPath";
+    await (async () => {
```

Comment:
@jagpreetsinghsasan Is this anonymous wrapper function needed? Does it not compile without it?

Reply:
Yes, its no longer needed. Fixed.